src/connection/host_lookup.c

```diff
@@ -0,0 +1,635 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * host_lookup.c
+ */
+
+#ident "$Id$"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <string.h>
+#include <limits.h>
+#include <errno.h>
+#include <unordered_map>
+
+#if defined (WINDOWS)
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#else
+#include <arpa/inet.h>
+#include <netdb.h>
+#endif
+
+#include "system_parameter.h"
+#include "environment_variable.h"
+#include "message_catalog.h"
+
+#define HOSTNAME_LEN                 (256)
+#define MAX_NUM_HOSTS                (256)
+#define LINE_BUF_SIZE                (512)
+#define IPADDR_LEN                   (17)
+#define NUM_IPADDR_DOT               (3)
+#define IPv4_ADDR_LEN                (4)
+
+#define NUM_DIGIT(VAL)              (size_t)(log10 (VAL) + 1)
+#define FREE_MEM(PTR)           \
+        do {                    \
+          if (PTR) {            \
+            free(PTR);          \
+            PTR = 0;    \
+          }                     \
+        } while (0)
+
+#define FREE_HOSTENT_MEM(PTR)        \
+        do {                    \
+          if (PTR) {            \
+            FREE_MEM (PTR->h_aliases[0]);       \
+            FREE_MEM (PTR->h_addr_list[0]);     \
+            FREE_MEM (PTR->h_addr_list);                \
+            FREE_MEM (PTR->h_aliases);          \
+            FREE_MEM (PTR->h_name);             \
+            FREE_MEM (PTR);                     \
+           }                                    \
+        }while(0)
+
+typedef enum
+{
+  HOSTNAME_TO_IPADDR = 0,
+  IPADDR_TO_HOSTNAME = 1,
+} LOOKUP_TYPE;
+
+typedef enum
+{
+  INSERT_IPADDR = 0,
+  INSERT_HOSTNAME = 1,
+} HOSTENT_INSERT_TYPE;
+
+typedef enum
+{
+  USE_GLIBC_HOSTS = 0,
+  USE_USER_DEFINED_HOSTS = 1,
+} HOST_LOOKUP_TYPE;
+
+typedef enum
+{
+  LOAD_FAIL = -1,
+  LOAD_INIT,
+  LOAD_SUCCESS,
+} HOSTS_CONF_LOAD_STATUS;
+
+static const char *user_hosts_File = "hosts.conf";
```

Comment:
How about the code below?
#define USER_HOSTS_FILE   "hosts.conf"

Reply:
Actually, there are many cases that pass the 'filename' parameter in envvar_vardir_file().
(envvar_vardir_file (char *path, size_t size, const char *filename))

So, I think macro also does not matter, I will change it.